src/main/java/fi/otavanopisto/kuntaapi/server/integrations/linkedevents/LinkedEventsEventProvider.java

```diff
@@ -0,0 +1,221 @@
+    if (!identifierRelationController.isChildOf(eventId, attachmentId)) {
+      return null;
+    }
+    
+    AttachmentData imageData = getImageData(organizationId, attachmentId);
+    if (size != null) {
+      return scaleEventImage(imageData, size);
+    }
+    
+    return imageData;
+  }
+
+  private AttachmentData getImageData(OrganizationId organizationId, AttachmentId attachmentId) {
+    AttachmentData storedAttachmentData = getStoredAttachmentData(attachmentId);
+    if (storedAttachmentData != null) {
+      return storedAttachmentData;
+    }
+    
+    return downloadImageData(organizationId, attachmentId);
+  }
+  
+  private AttachmentData getStoredAttachmentData(AttachmentId attachmentId) {
+    StoredBinaryData storedBinaryData = linkedEventsAttachmentDataResourceContainer.get(attachmentId);
+    if (storedBinaryData != null) {
+      try {
+        return new AttachmentData(storedBinaryData.getContentType(), IOUtils.toByteArray(storedBinaryData.getDataStream()));
+      } catch (IOException e) {
+        logger.log(Level.SEVERE, "Failed to read stream data", e);
+      } 
+    }
+     
+    return null;
+  }
+
+  private boolean isWithinTimeRanges(Event event, OffsetDateTime startBefore,
+      OffsetDateTime startAfter, OffsetDateTime endBefore, OffsetDateTime endAfter) {
+    
+    OffsetDateTime eventStart = event.getStart();
+    OffsetDateTime eventEnd = event.getEnd();
+    
+    if (eventStart == null || eventEnd == null) {
+      return false;
+    }
+    
+    if (!isWithinTimeRange(startBefore, startAfter, eventStart)) {
+      return false;
+    }
+    
+    if (!isWithinTimeRange(endBefore, endAfter, eventEnd)) {
+      return false;
+    }
+    
+    return true;
+  }
+
+  private boolean isWithinTimeRange(OffsetDateTime before, OffsetDateTime after, OffsetDateTime time) {
+    if (before != null && time.isAfter(before)) {
+      return false;
+    }
+
+
+    if (after != null && time.isBefore(after)) {
+      return false;
+    }
+    
+    return true;
+  }
+  
+  private AttachmentData scaleEventImage(AttachmentData imageData, Integer size) {
+    if (imageData == null) {
+      return null;
+    }
+    
+    BufferedImage bufferedImage = imageReader.readBufferedImage(imageData.getData());
+    if (bufferedImage != null) {
+      String formatName = imageWriter.getFormatName(imageData.getType());
+      BufferedImage scaledImage = imageScaler.scaleToCover(bufferedImage, size, true);
+      byte[] scaledImageData = imageWriter.writeBufferedImage(scaledImage, formatName);
+      if (scaledImageData != null) {
+        String contentType = imageWriter.getContentTypeForFormatName(formatName);
+        return new AttachmentData(contentType, scaledImageData);
+      }
+    }
+    
+    return null;
+  }
+  
+  private AttachmentData downloadImageData(OrganizationId organizationId, AttachmentId imageId) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "organizationId". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)